### DISCUSSION TITLE
Option to allow requests ONLY from currently running scheduled playlist(s)

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6125] [Ideas]
I’m currently on Rolling Release #[5cbf9c9](https://github.com/AzuraCast/AzuraCast/commit/5cbf9c91cae1a234c428cf43d769fe2349e05bd9) (2023-02-28 8:15), and the automatic restriction of song requests works rather fine, in principle.

I have 3 weighted playlists feeding the general rotation, plus lots of scheduled playlists to make for a useful day programming, i.e., start the day with some Pop, play some Easy Listening around midday, and turn to 2-hour genre-based "show blocks" in the evening, which are filled by playlists if no DJ takes over and does a live show.

Now song requests will allow selecting from whatever is in general rotation (the 3 playlists mentioned: Classic Rock, Hard Rock, Folk Rock) _plus_ whatever is in a scheduled playlist (say: EBM). So my listeners can request Classic Rock, Hard Rock and Folk Rock _plus_ EBM titles during an EBM "show block"—not so nice if you want to allow requests but they should "fit in".

It’d be just great if we had **an extra option in scheduled playlists like "[X] Allow requests _only_ from running schedule/this playlist"**.

It _should_ take into account that you might run 2 scheduled playlists _at the same time_ (say: Krautrock + Prog Rock), whose weights are used to fill the currently scheduled block. In this case, requests should only be honored from the playlist(s) that have "[X] Allow requests _only_…" set, and whatever is (theoretically) allowed in general rotation disregarded/not requestable.

This construct would allow listener requests fitting in with the general rotation during "the rest of the day", as well as restricting requests to scheduled "show" blocks. A rather useful "day programming" with no off-tracks.

**Comment by BusterNeece** [comment]
This should be a feature request.

**Comment by Moonbase59** [comment]
Oops, sorry. Reposted an excerpt (linking to here because of text length) at https://features.azuracast.com/suggestions/366854/option-to-allow-requests-only-from-currently-running-scheduled-playlists
